tools/tpm2_flushcontext.c

```diff
@@ -250,17 +288,21 @@ static bool on_option(char key, char *value) {
     switch (key) {
     case 't':
         ctx.property[ctx.encountered_option] = TPM2_TRANSIENT_FIRST;
+        ctx.encountered_option++;
         break;
     case 'l':
         ctx.property[ctx.encountered_option] = TPM2_LOADED_SESSION_FIRST;
+        ctx.encountered_option++;
         break;
     case 's':
         ctx.property[ctx.encountered_option] = TPM2_ACTIVE_SESSION_FIRST;
+        ctx.encountered_option++;
```

Comment:
This isn't a bug with this patch, but this needs bounds checking... ill submit a patch since I made the bug:
```bash
tpm2_flushcontext -tttttttt
Segmentation fault (core dumped)

Reply:
okay I will merge and rebase on that PR.